### PR TITLE
[Merged by Bors] - perf(VectorBundle/FiberwiseLinear): speed up

### PR DESCRIPTION
**Pull request by grunweg**
#12061 regressed this file pretty badly; speed it up again by mostly reverting a change from that PR, and pushing it further: instead of applying `mem_iUnion` (which is slow), extract a more specialised statement `mem_aux` and use it.

Before these changes, the `profiler` output on `smoothFiberwiseLinear` was
```
simp took 2.01s
tactic execution of Lean.Parser.Tactic.rintro took 103ms
simp took 905ms
simp took 507ms
simp took 1.26s
simp took 1.14s
elaboration took 645ms
```
after these changes, it is
```
simp took 534ms
simp took 183ms
simp took 127ms
simp took 421ms
simp took 365ms
elaboration took 350ms
```
Still not great at all, but deeper fixes require more time and expertise than I currently have.

---

I cannot recover the link to the post-merge perf. run from speed.lean-lang.org; help with that welcome.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
!bench

**Comment by urkud** [comment]
Do you know why `simp only [mem_iUnion]` is so slow?

**Comment by grunweg** [comment]
I do not. Understand and fixing this would be nice. Would you like me to add a TODO?

**Comment by urkud** [comment]
Yes, please. Otherwise LGTM. Thanks!
bors d+

File: Mathlib/Geometry/Manifold/VectorBundle/FiberwiseLinear.lean [hunk 1]
```diff
@@ -223,6 +223,21 @@ theorem SmoothFiberwiseLinear.locality_aux₂ (e : PartialHomeomorph (B × F) (B
 
 variable (F B IB)
 
+variable {F B IB} in
+-- Having this private lemma speeds up `simp` calls a lot.
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/VectorBundle/FiberwiseLinear.lean:227]:
```suggestion
-- Having this private lemma speeds up `simp` calls below a lot.
-- TODO: understand why and fix the underlying issue (relatedly, the `simp` calls
-- in `smoothFiberwiseLinear` are quite slow, even with this change)

**Comment by grunweg** [comment]
Thanks for the review!
bors r+

**Pull request closed without merging**